drkafka/src/main/java/com/pinterest/doctorkafka/modules/plugins/manager/DoctorKafkaPluginManager.java

```diff
@@ -0,0 +1,72 @@
+package com.pinterest.doctorkafka.modules.plugins.manager;
+
+import com.pinterest.doctorkafka.modules.plugins.AbstractPlugin;
+import com.pinterest.doctorkafka.modules.plugins.errors.PluginException;
+
+import org.apache.commons.configuration2.AbstractConfiguration;
+import org.apache.commons.configuration2.CompositeConfiguration;
+import org.apache.commons.configuration2.Configuration;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+
+/**
+ * DoctorKafkaPluginManager is a naive implementation of PluginManager.
+ * It will search for the plugin in the classpath and apply configuration to the plugins.
+ * Additional configurations can be applied and will have a higher priority compared to plugin
+ * configurations.
+ *
+ * Plugin config should be structured as the following:
+ * plugin-1.class=com.foo.bar.plugin1
+ * plugin-1.some.attribute=foo
+ * plugin-2.class=com.foo.bar.plugin2
+ * plugin-2.some.other.attribute=bar
+ */
+
+public class DoctorKafkaPluginManager implements PluginManager {
+  private static final Logger LOG = LogManager.getLogger(DoctorKafkaPluginManager.class);
+  private static final String PLUGIN_CONFIG_CLASS_KEY = "class";
+  private Map<String, Class> pluginMap = new ConcurrentHashMap<>();
+  private AbstractConfiguration pluginConfig;
+
+  public DoctorKafkaPluginManager(AbstractConfiguration pluginConfig){
+    this.pluginConfig = pluginConfig;
+  }
+
+  public AbstractPlugin getPlugin(String pluginName) throws Exception {
+    return getPlugin(pluginName, null);
+  }
+
+  @Override
+  public AbstractPlugin getPlugin(String pluginName, AbstractConfiguration additionalConfigs) throws Exception {
+    Configuration config = pluginConfig.subset(pluginName);
+    if (config == null) {
+      throw new PluginException("Could not find config for plugin: "+ pluginName);
+    }
+    String pluginClass = config.getString(PLUGIN_CONFIG_CLASS_KEY);
```

Comment:
Class loading could also be done using autoscan via reflection. Plugin class would need to implement a plugin name method. E.g. https://github.com/srotya/sidewinder/blob/development/core/src/main/java/com/srotya/sidewinder/core/functions/list/FunctionTable.java